src/api/axios.ts

```diff
@@ -62,14 +66,21 @@ const fallbackForArray = async (
 export const fallbackToPublic = (
   request: () => Promise<AxiosResponse>,
   publicRequest: () => Promise<AxiosResponse>,
-  fallbackData?: unknown,
+  options?: FallbackToPublicOptions,
 ) => {
-  const isAuthenticated = isUserAuthenticated();
+  let isAuthenticated;
+
+  // if the call should be public, override isAuthenticated
+  if (options?.public) {
+    isAuthenticated = false;
+  } else {
+    isAuthenticated = isUserAuthenticated();
+  }
```

Comment:
Maybe this can be simplified to `const isAuthenticated = options?.public ? false : isUserAuthenticated()`?

Reply:
Indeed it can work! But I prefer to keep it separated, I think it's clearer.